src/layman/rest_publication_test.py

```diff
@@ -46,3 +48,69 @@ def check_response(resp):
     r = patch_method(workspace, publication)
 
     delete_method(workspace, publication)
+
+
+class TestSoapClass:
+    username = 'test_rest_soap_user'
+    publ_name_prefix = 'test_rest_soap_'
+    authz_headers = process_client.get_authz_headers(username)
+    access_rights_rowo = {'read': f"{username}", 'write': f"{username}"}
+    access_rights_rewo = {'read': f"{username},EVERYONE", 'write': f"{username}"}
+    access_rights_rewe = {'read': f"{username},EVERYONE", 'write': f"{username},EVERYONE"}
+
+    @pytest.fixture(scope='class')
+    def reserve_username(self):
+        process_client.reserve_username(self.username, headers=self.authz_headers)
+        yield
+
+    @pytest.mark.usefixtures('liferay_mock', 'ensure_layman', 'reserve_username')
+    @pytest.mark.parametrize('params_and_expected_list', [
+        [(access_rights_rowo, False), (access_rights_rewe, True)],
+        [(access_rights_rewo, True)],
+        [(access_rights_rewe, True), (access_rights_rowo, False)],
+    ])
+    @pytest.mark.parametrize('publ_type', [
+        'layer',
+        'map',
+    ])
+    def test_soap_authz(self, publ_type, params_and_expected_list):
+        username = self.username
+        publ_name_prefix = self.publ_name_prefix
+        authz_headers = self.authz_headers
+
+        post_method = {
```

Comment:
Prosím dát do process_client. Buď jako takovéto slovníky nebo metody typu publish_publication a typem publikace jako parametrem.